### PR TITLE
Fixed navigation bar color at private tab

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
Resolves https://github.com/brave/brave-browser/issues/41545

This PR fixes issue with the navigation bar color for private tab.

Color is used from  @simoarpe 's upcoming PR https://github.com/brave/brave-core/pull/25850/commits/c972bdd57b625ba10fb3877979d909fe52059936 

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Screenshot&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|image
--|--

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Take Nightly 1.72.66
2. Open private tab

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/tabbed_mode/BraveTabbedNavigationBarColorControllerBase.java [hunk 1]
```diff
@@ -34,7 +37,11 @@ public int getNavigationBarColor(boolean forceDarkNavigationBar) {
         if (BottomToolbarConfiguration.isBottomToolbarEnabled()
                 && BraveMenuButtonCoordinator.isMenuFromBottom()
                 && mContext != null) {
-            return mContext.getColor(R.color.default_bg_color_baseline);
+            if (mTabModelSelector.isIncognitoSelected()) {
```
Comment (samartnik) [inline android/java/org/chromium/chrome/browser/tabbed_mode/BraveTabbedNavigationBarColorControllerBase.java:40]: We should think of how to detect whether they deleted/renamed this variable in the upstream. Same applies to `mContext`. In any case we should treat variable that we deleted in the bytecode from the upstream's code as `Nullable`.

**Review by samartnik — APPROVED** [review]
lgtm

**Review by simoarpe — APPROVED** [review]
👍 Changes look good!
Thank you for fixing this!

**Review by deeppandya — APPROVED** [review]
LGTM. Thank you @AlexeyBarabash

**Comment by hffvld** [comment]
Verified on `Pixel 7` using version(s):
```
Device/OS: Pixel 7 / panther_beta-user 15 AP41.240823.009 release-keys
Brave build: 1.73.7 
Chromium: 130.0.6723.44 (Official Build) canary (64-bit) 
```
---------------------------

**STEPS:**
1. Make sure that the device's `Dark mode` is OFF
2. Launch Brave > *Confirm that the `Light mode` UI is shown*
3. Open `Private tab` > *Verify*

**ACTUAL RESULTS:**
- Verified that the `OS navigation bar` in the `Private tab` has the correct color.
- Verified the same when the device theme is set to `Light` and `Dark` mode.
- Verified the same for the `Standard tab` as well.
- Verified the same when Brave `theme` in `Settings` -> `Appearance` is set to `System default`, `Light` and `Dark` mode.
- Verified the same for `Gesture navigation` and `3-button navigation`.

---------------------------

`Reproduced in Nightly 1.73.2`

|1|2|
|:-:|:-:|
|  |  |

`Verified in Nightly 1.73.7`

|1|2|3|4|
|:-:|:-:|:-:|:-:|
|  |  |  |  |
|  |  |  |  |

**Pull request merged**